src/core/io/src/4C_io_input_spec_builders.hpp

```diff
@@ -0,0 +1,882 @@
+     *      entry<double>("gamma"),
+     *   },
+     *   {.required = false}
+     *   );
+     *
+     * //Groups may be nested
+     * group("outer",
+     *  {
+     *    entry<int>("a"),
+     *    group("inner",
+     *    {
+     *      entry<double>("b"),
+     *      entry<std::string>("c"),
+     *    }
+     *    ),
+     *  });
+     *
+     * @endcode
+     *
+     * A group introduces a new scope in the input. This group scope is not only useful for
+     * structuring the input file, but also to selectively activate a set of parameters, as
+     * demonstrated in the second example. If an optional group is not present in the input, none of
+     * the entries are required. If the group is present, all entries are required. This is often
+     * exactly what you need: a group activates a feature which requires certain parameters to
+     * be present.
+     *
+     * Whether a group has a default value is determined by the default values of its entries. If
+     * all entries have default values, the group implicitly has a default value. In this case, if
+     * the group is optional and not present in the input, the default values of the entries are
+     * stored under the group name in the container.
+     *
+     * @note If you want to group multiple InputSpecs without creating a new scope, use the other
+     * group() function which does not take a name.
+     *
+     */
+    [[nodiscard]] InputSpec group(
+        std::string name, std::vector<InputSpec> specs, GroupData data = {});
+
+    /**
+     * A group of multiple InputSpecs. In contrast to the other group() function, this function
+     * produces a group without a name and, consequently, no scope is created in the input.
+     * Example:
+     *
+     * @code
+     * group({
+     *   entry<int>("a"),
+     *   entry<double>("b"),
+     *   entry<std::string>("c"),
+     *   });
+     * @endcode
+     *
+     * This functions gathers multiple InputSpecs on the same level and treats them as a single
+     * InputSpec.
+     */
+    [[nodiscard]] InputSpec group(std::vector<InputSpec> specs);
+
+    /**
+     * Exactly one of the given InputSpecs is expected. There is no support for default values and
+     * one of the InputSpecs is always required. For example, the example shown for the group()
+     * function may be enhanced to require exactly one of two groups:
+     *
+     * @code
+     * one_of({
+     *  group("OneStepTheta",
+     *  {
+     *    entry<double>("theta"),
+     *  }),
+     *  group("GenAlpha",
+     *  {
+     *    entry<double>("alpha_f"),
+     *    entry<double>("alpha_m"),
+     *    entry<double>("gamma"),
+     *  }),
+     *  });
+     * @endcode
+     *
+     * This InputSpecs requires either the "OneStepTheta" group or the "GenAlpha" group to be
+     * present in the input. If both or none of them are present, an exception is thrown.
+     */
+    [[nodiscard]] InputSpec one_of(std::vector<InputSpec> specs);
```

Comment:
This is similar to the `SwitchComponent`, right?

Reply:
Exactly. `one_of` can be used to achieve what `SwitchComponent` does.